frontend/src/metabase/dashboard/components/Dashboard/Dashboard.unit.spec.tsx

```diff
@@ -0,0 +1,188 @@
+import { renderWithProviders, waitForLoaderToBeRemoved } from "__support__/ui";
+import {
+  createMockDashboardCard,
+  createMockDashboard,
+} from "metabase-types/api/mocks";
+import type { Dashboard, Parameter } from "metabase-types/api";
+import { Dashboard as DashboardComponent } from "./Dashboard";
+
+type SetupOpts = {
+  dashboardId: number | null;
+  dashboard: Dashboard | null;
+  selectedTabId: number | null; // when there's no tabs, is null
+  parameters: Parameter[]; // when empty, is an empty array
+  skipLoader?: boolean;
+  isEditing: boolean;
+};
+
+function dashboardWithCards(first: { isDirty?: boolean } = {}) {
+  return createMockDashboard({
+    dashcards: [
+      createMockDashboardCard({ id: 1, isDirty: first.isDirty } as any),
+      createMockDashboardCard({ id: 2 }),
+    ],
+  });
+}
+
+async function setup(overrides: Partial<SetupOpts> = {}) {
+  const mockDashboard = createMockDashboard({ id: 10 }); // value is irrelevant
+
+  const opts: SetupOpts = {
+    dashboardId: 10,
+    dashboard: mockDashboard,
+    selectedTabId: null,
+    parameters: [],
+    isEditing: false,
+    ...overrides,
+  };
+  const mockLoadDashboardParams = jest.fn();
+  const mockFetchDashboard = jest.fn(() => mockDashboard);
+  const mockFetchDashboardCardData = jest.fn();
+  const mockFetchDashboardCardMetadata = jest.fn();
+
+  function TestComponent(props: SetupOpts) {
+    return (
+      <DashboardComponent
+        dashboard={props.dashboard}
+        dashboardId={props.dashboardId}
+        parameters={props.parameters}
+        loadDashboardParams={mockLoadDashboardParams}
+        fetchDashboard={mockFetchDashboard}
+        fetchDashboardCardData={mockFetchDashboardCardData}
+        fetchDashboardCardMetadata={mockFetchDashboardCardMetadata}
+        isEditing={props.isEditing}
+        // stuff below doesn't change
+        parameterValues={undefined}
+        location={global.location}
+        isAdmin={false}
+        isFullscreen={false}
+        isNightMode={false}
+        isSharing={false}
+        isEditingParameter={false}
+        isNavbarOpen={false}
+        isHeaderVisible={false}
+        isAdditionalInfoVisible={false}
+        isNavigatingBackToDashboard={false}
+        editingOnLoad={false}
+        addCardOnLoad={null}
+        isAutoApplyFilters={false} // TODO true or false?
```

Comment:
Think we should use `dashboard?.auto_apply_filters` like [here](https://github.com/metabase/metabase/blob/fea5d8cb4dd3e8b8a25bc15d5b15fca36ff8ff04/frontend/src/metabase/dashboard/selectors/selectors.js#L157)

Reply:
Done!